src/components/trail.js

```diff
@@ -6,58 +6,265 @@ AFRAME.registerComponent('trail', {
   },
 
   init: function () {
-    var geometry = this.geometry = new THREE.BufferGeometry();
-    var maxPoints = this.maxPoints = 12;
-    var vertices = this.vertices = new Float32Array(36 * maxPoints);
-    var colors = this.colors = new Float32Array(48 * maxPoints);
-    var bladeColor = this.bladeColor = new THREE.Color(this.data.color);
-
-    this.bladeColor = {
-      red: bladeColor.r,
-      green: bladeColor.g,
-      blue: bladeColor.b,
-      alpha: 1.0
-    };
+    //TRAIL CONFIG ---------------------------------------------------------------------
+    //You must call init (and potentially dispose already existing mesh) after any config change
+    this.trailType = 'bright' //available types: 'bright', 'dim', 'acc'
+
+    this.lifetime = 20; //frames
+    this.verticalResolution = 120; //quads
+    this.horizontalResolution = 2; //quads
+
+    this.brightTrailWidth = 0.9; //meters
+    this.dimTrailWidth = 0.3; //meters
+    this.accTrailHalfWidth = 0.05; //meters
+    //TRAIL CONFIG ---------------------------------------------------------------------
+
     this.saberEl = this.el.querySelector('.blade');
+    this.bladeColor = new THREE.Color(this.data.color);
```

Comment:
Values in 'data' can be changed through binding. Add `bladeColor` update in `update()`